homework/index.js

```diff
@@ -29,26 +39,142 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function appendRepoDetail(header, description, parentTable) {
+    const repoDetailRow = createAndAppend('tr', parentTable);
+    createAndAppend('th', repoDetailRow, { text: header });
+    createAndAppend('td', repoDetailRow, { text: description });
+    return repoDetailRow;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
-          text: err.message,
+  function getDateTimeText(dateTimeStr) {
+    const dateTime = new Date(dateTimeStr);
+    return dateTime.toLocaleString();
+  }
+
+  function sortRepositoriesByNameAscending(firstRepo, secondRepo) {
+    return firstRepo.name.localeCompare(secondRepo.name);
+  }
+
+  function populateSelectElement(repositories, select) {
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function renderRepoDetails(repo, section) {
+    const repoDetailsTable = createAndAppend('table', section, {
+      class: 'repository-item-table',
+    });
+    const firstRow = appendRepoDetail('Repository:', '', repoDetailsTable);
+    createAndAppend('a', firstRow.lastChild, {
+      href: repo.html_url,
+      text: repo.name,
+      target: '_blank',
+    });
+    appendRepoDetail('Description:', repo.description, repoDetailsTable);
+    appendRepoDetail('Forks:', repo.forks, repoDetailsTable);
+    appendRepoDetail(
+      'Updated:',
+      getDateTimeText(repo.updated_at),
+      repoDetailsTable,
+    );
+  }
+
+  function renderContributorDetails(contributor, ul) {
+    const repoContributorListItem = createAndAppend('li', ul, {
+      class: 'contributors-item',
+    });
+    createAndAppend('img', repoContributorListItem, {
+      src: contributor.avatar_url,
+      alt: 'Contributor avatar picture',
+    });
+    createAndAppend('a', repoContributorListItem, {
+      href: contributor.html_url,
+      text: contributor.login,
+      target: '_blank',
+    });
+    createAndAppend('span', repoContributorListItem, {
+      text: contributor.contributions,
+    });
+  }
+
+  function createHeaderSection(title, parent) {
+    const header = createAndAppend('header', parent);
+    createAndAppend('span', header, {
+      text: title,
+      class: 'header-title',
+    });
+    return createAndAppend('select', header);
+  }
+
+  function createMainSectionWithSubSections(parent) {
+    const mainSection = createAndAppend('main', parent, { class: 'container' });
+    dom[DETAIL_SECTION] = createAndAppend('section', mainSection, {
+      class: DETAIL_SECTION,
+    });
```

Comment:
I doubt the added value of using these CONSTANTS. I would myself hard code the class name and use dot notation for the property names (it also reduced the amount of SHOUTING in your code 😉):

```js
dom.detailSection = createAndAppend('section', mainSection, {
  class: 'repository-detail-container',
});
```